bundle/config/validate/interpolation_in_auth_config.go

```diff
@@ -0,0 +1,77 @@
+package validate
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/auth"
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+	"github.com/databricks/cli/libs/dyn/dynvar"
+)
+
+type noInterpolationInAuthConfig struct{}
+
+func NoInterpolationInAuthConfig() bundle.Mutator {
+	return &noInterpolationInAuthConfig{}
+}
+
+func (f *noInterpolationInAuthConfig) Name() string {
+	return "validate:interpolation_in_auth_config"
+}
+
+func (f *noInterpolationInAuthConfig) Apply(ctx context.Context, b *bundle.Bundle) diag.Diagnostics {
+	authFields := []string{
+		// Generic attributes.
+		"host", "profile", "auth_type", "metadata_service_url",
+
+		// OAuth specific attributes.
+		"client_id",
+
+		// Google specific attributes.
+		"google_service_account",
+
+		// Azure specific attributes.
+		"azure_resource_id", "azure_use_msi", "azure_client_id", "azure_tenant_id",
+		"azure_environment", "azure_login_app_id",
```

Comment:
For future: please use newlines to separate entries in a list. Diffs will be easier to read.

Reply:
Done in https://github.com/databricks/cli/pull/2412